### ISSUE TITLE
bug(dev-cli): Unexpected wasm memory growth 

### ISSUE THREAD

**Issue by PeterFarber** [permaweb/ao#915]
## Description
When running wasm64 (WebAssembly 64-bit) modules in our application, we've encountered an issue where memory usage steadily increases over time. This behavior occurs unpredictably and seems to be independent of specific module functions or operations.

## Expected Behavior
Memory usage should remain stable or increase within expected limits as defined by the application's memory management and garbage collection routines.

## Actual Behavior
Memory consumption grows continuously, eventually leading to performance degradation and potential crashes due to memory exhaustion.

## Steps to Reproduce (uncertain)
Unfortunately, the issue of unexpected wasm64 memory growth appears sporadic, making it challenging to provide precise steps for reproduction.

## Impact
This issue severely impacts the stability and performance of our application when running wasm64 modules, especially in long-running sessions.

## Proposed Solution
Investigate and address the root cause of the memory growth in wasm64 modules. This may involve:
- Modify Emscripten build memory configuration flags.
- Reviewing memory management strategies within the wasm64 module.
- Checking for potential memory leaks or inefficient memory handling.
- Ensuring compatibility with the WebAssembly runtime.

**Comment by ppedziwiatr** [comment]
I was recently analysing the memory dump of our token and was kinda suprised that that it has almost 90mb - where effectively sth like a map of ~100 is being hold - BUT - we've changed its source code several times (if I understand correctly -  `aos load` sends an `Eval` interaction to the AOS process - and it causes lua's `load` to interpret the new code) - so maybe it has sth to code with loading/updated process' code?

**Comment by PeterFarber** [comment]
> I was recently analysing the memory dump of our token and was kinda suprised that that it has almost 90mb - where effectively sth like a map of ~100 is being hold - BUT - we've changed its source code several times (if I understand correctly - `aos load` sends an `Eval` interaction to the AOS process - and it causes lua's `load` to interpret the new code) - so maybe it has sth to code with loading/updated process' code?
> 
> Another example - our oracle process - https://cu.ao-testnet.xyz/state/fev8nSrdplynxom78XaQ65jSo7-88RxVVVPwHG8ffZk - 50mb
> 
> This is _NOT_ a AOS process - but a process built with `dev-cli` from this source code https://github.com/warp-contracts/ao-redstone-oracle/blob/main/redstone-oracle-process/process.lua
> 
> Quickly review the contents of the dump - it seems that like 60% are the source codes and the rest is the state of the process (in this case some jsons stored in a lua table) <img alt="Zrzut ekranu 2024-07-30 o 17 17 03" width="1512" src="https://private-user-images.githubusercontent.com/84311757/353518597-213832c4-48da-4f80-a5b7-512386d25e25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0MzY4MjgsIm5iZiI6MTcyMjQzNjUyOCwicGF0aCI6Ii84NDMxMTc1Ny8zNTM1MTg1OTctMjEzODMyYzQtNDhkYS00ZjgwLWE1YjctNTEyMzg2ZDI1ZTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMxVDE0MzUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5ODBhZjMzZDlkMzdiOTAxOGRjYmI2NjIzYWUwMTA3MWI4ZDcwZmZmMTY2N2U1ZGRkNjNhMjAzNDQ0NGI2YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9BIQgRdpMAbgXKjibZeOky4NDEiUzYh4huyDPuZPE8E">

Yeah looks like most of the dump is that the returned memory is never being freed. Do you have discord? Would like to review your finding.

**Comment by ppedziwiatr** [comment]
We have a dedicated Slack with you Guys - https://redstone-ujk1058.slack.com/archives/C06NG8C2CNR - so maybe drop us a message there?
